mitmproxy/proxy/layers/http/_http_h2.py

```diff
@@ -49,9 +49,19 @@ class BufferedH2Connection(h2.connection.H2Connection):
 
     def __init__(self, config: h2.config.H2Configuration):
         super().__init__(config)
+        self.local_settings.initial_window_size = 2**24
+        self.local_settings.max_frame_size = 2**17
+        self.max_inbound_frame_size = 2**17
+        # hyper-h2 pitfall: we need to acknowledge here, otherwise its sends out the old settings.
+        self.local_settings.acknowledge()
         self.stream_buffers = collections.defaultdict(collections.deque)
         self.stream_trailers = {}
 
+    def initiate_connection(self):
+        super().initiate_connection()
+        # The connection flow-control window can only be changed using WINDOW_UPDATE frames.
```

Comment:
Let's make it a bit more clear here that this supplements the setting (I had to re-read the RFC to be reminded of that second separate window):
```suggestion
        # We increase the flow-control window for new streams with a setting,
        # but we need to increase the overall connection flow-control window as well.
```